node/libs/concurrency/src/sync/prunable_mpsc/tests.rs

```diff
@@ -0,0 +1,94 @@
+use crate::ctx;
+use assert_matches::assert_matches;
+use tokio::time::{timeout, Duration};
+
+// Test scenario:
+// 1. Pre-send two sets of 1000 values, so that the first set is expected to be pruned.
+// 2. Send a third set of 1000 values in parallel to receiving.
+#[tokio::test]
+async fn test_prunable_mpsc() {
+    let ctx = ctx::test_root(&ctx::RealClock);
+
+    #[derive(Debug, Clone)]
+    struct ValueType(usize, usize);
+
+    #[allow(clippy::type_complexity)]
+    let (send, mut recv): (
+        super::Sender<ValueType, Result<(), usize>>,
+        super::Receiver<ValueType, Result<(), usize>>,
+    ) = super::channel(|a: &ValueType, b: &ValueType| {
+        // Prune values with the same i.
+        a.1 == b.1
+    });
+
+    let res: Result<(), ctx::Canceled> = crate::scope::run!(&ctx, |ctx, s| async move {
+        // Pre-send sets 0 and 1, 1000 values each.
+        // Set 0 is expected to be pruned and dropped.
+        let values = (0..2000).map(|i| ValueType(i / 1000, i % 1000));
+        for val in values {
+            let res_recv = send.send(val.clone()).await;
+            s.spawn(async move {
```

Comment:
nit: "async move" is rarely applicable in scope::run/spawn/spawn_bg. The convention is to always use just "async { .. }" which is way more flexible than "async move". If you need to move x explicitly inside the closure, use "let x = x;". If x implements Copy, first wrap it in NoCopy then move it as previously.

Reply:
Thanks, working now as you suggested.